installer/dual_rootfs_device.go

```diff
@@ -192,13 +220,23 @@ func (d *dualRootfsDeviceImpl) getInactivePartition() (string, string, error) {
 	if err != nil {
 		return "", "", errors.New("Error obtaining inactive partition: " + err.Error())
 	}
+	return d.getPartitionImpl(inactivePartition)
+}
+
+func (d *dualRootfsDeviceImpl) getActivePartition() (string, string, error) {
+	activePartition, err := d.GetActive()
+	if err != nil {
+		return "", "", errors.New("Error obtaining inactive partition: " + err.Error())
+	}
+	return d.getPartitionImpl(activePartition)
+}
 
-	log.Debugf("Marking inactive partition (%s) as the new boot candidate.", inactivePartition)
+func (d *dualRootfsDeviceImpl) getPartitionImpl(partition string) (string, string, error) {
 
-	partitionNumberDecStr := inactivePartition[len(strings.TrimRight(inactivePartition, "0123456789")):]
+	partitionNumberDecStr := partition[len(strings.TrimRight(partition, "0123456789")):]
 	partitionNumberDec, err := strconv.Atoi(partitionNumberDecStr)
 	if err != nil {
-		return "", "", errors.New("Invalid inactive partition: " + inactivePartition)
+		return "", "", errors.New("Invalid inactive partition: " + partition)
```

Comment:
Should this now say "Invalid partition"?

Reply:
Yes, agreed!